### PR TITLE
Fix another case of non-ready set.

### PR DESCRIPTION
**Pull request by KochetovNicolai**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix `Not-ready Set is passed as the second argument for function 'in'` exception for query `WITH TOTALS`, `JOIN` and `IN (subquery)`. https://github.com/ClickHouse/ClickHouse/issues/52991

Set was not created for main query execution but was required for totals part. Now, creating sets is delaying totals execution as well.

```sql
SELECT *
FROM
(
    SELECT item_id
    FROM tab
    GROUP BY item_id
        WITH TOTALS
    ORDER BY '922337203.6854775806' IN (
            SELECT NULL
        ) ASC
) AS l
RIGHT JOIN
(
    SELECT item_id
    FROM tab
) AS r ON l.item_id = r.item_id
WHERE NULL

```

### CODE REVIEW

**Comment by robot-ch-test-poll** [comment]
*This is an automated comment for commit 6d9e7f98a1c572e847f1acfa2ac765f85a2529f4 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/57423/6d9e7f98a1c572e847f1acfa2ac765f85a2529f4/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td><a href="https://github.com/ClickHouse/ClickHouse/actions/runs/7060766738">✅ success</a></td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/57423/6d9e7f98a1c572e847f1acfa2ac765f85a2529f4/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

File: src/QueryPipeline/QueryPipelineBuilder.cpp [hunk 1]
```diff
@@ -602,7 +602,12 @@ void QueryPipelineBuilder::addPipelineBefore(QueryPipelineBuilder pipeline)
...
 
-    IProcessor::PortNumbers delayed_streams(pipe.numOutputPorts());
+    pipeline.dropTotalsAndExtremes();
+
+    bool has_totals = pipe.getTotalsPort();
+    bool has_extremes = pipe.getExtremesPort();
+    size_t num_extra_ports = (has_totals ? 1 : 0) + (has_extremes ? 1 : 0);
+    IProcessor::PortNumbers delayed_streams(pipe.numOutputPorts() + num_extra_ports);
     for (size_t i = 0; i < delayed_streams.size(); ++i)
         delayed_streams[i] = i;
```
Comment (novikd) [inline src/QueryPipeline/QueryPipelineBuilder.cpp:612]: `std::iota`

**Review by novikd — APPROVED** [review]

**Comment by KochetovNicolai** [comment]
Fuzzer: https://github.com/ClickHouse/ClickHouse/pull/57540

**Comment by KochetovNicolai** [comment]
https://github.com/ClickHouse/ClickHouse/pull/57614

**Comment by KochetovNicolai** [comment]
oops, was merged by mistake
revert it if needed

**Pull request merged**